lib/src/themeoptions.dart

```diff
@@ -0,0 +1,17 @@
+class ThemeOptions {
+  final String? name;
+  final Map<String, dynamic>? custom;
+
+  const ThemeOptions({this.name, this.custom});
+
+  Map<String, dynamic> toJson() {
+    final result = <String, dynamic>{};
+    if (name != null) {
+      result['name'] = name;
+      result['custom'] = custom;
```

Comment:
custom could be null now, so I'd make 2 separate and unrelated `if`s, potentially leading to an empty object if both are null.